lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py

```diff
@@ -146,14 +177,20 @@
             resource_group: Testing
             virtual_network_name: vnet001
             subnet_name: subnet001
+            ip_configurations:
```

Comment:
Is this really the minimal parameters now, seems weird that you need to specify `public_ip_address_name` and `primary`